install-4.2.x.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
```

Comment:
Is there anything you found that you were not able to do in a `RUN` command that you needed to do that you couldn't in the Dockerfile?  I only ask out of a personal preference to keep things as simple as possible and keep everything in one place if at all possible (the Dockerfile).  It seems like the only thing that is new/different is with arrays and some checking of environment variables.  I'm just curious if it is worth adding perceived complexity for little gain to the end process.

Reply:
Not really, my main motivation was that I tend to see no added complexity and better readability.

The RUN command is actually a real shell script, which I find easier to read in a separate `.sh` file :
- allows syntax highlighting in most text editors
- allows shellcheck warnings
- allows use of variables (here, mainly `OMADA_DIR`)
- allows inserting of empty lines for better readability
- (optionally) perhaps a bit easier to extend if more complexity is needed in the future, if TP-Link adds some in its own `install.sh` 

That said, it is a matter of taste, and if you prefer keep it all in the `Dockerfile` I can do it of course.
In this case I would write something like this :

```dockerfile
RUN \
  echo "**** Install Dependencies ****" &&\
  apt-get update &&\
  apt-get install --no-install-recommends -y gosu net-tools openjdk-8-jre-headless tzdata wget &&\
  if [ "$ARCH" == "armv7l" ]; then \
    apt-get install --no-install-recommends -y mongodb \
  else \
    apt-get install --no-install-recommends -y mongodb-server-core \
  fi &&\
 ...
```
Let me know what you prefer.

Reply:
After thinking it over, I'm fine with an `install.sh` script.  At the end of the day, it's no different.